module/metrics/transaction.go

```diff
@@ -277,5 +280,8 @@ func (tc *TransactionCollector) TransactionExpired(txID flow.Identifier) {
 }
 
 func (tc *TransactionCollector) UpdateExecutionReceiptMaxHeight(height uint64) {
-	tc.maxReceiptHeight.Set(float64(height))
+	if height > tc.maxReceiptHeightValue {
```

Comment:
would this be called concurrently?

Also consider using this concurrent safe module
https://github.com/onflow/flow-go/blob/master/engine/consensus/sealing/counters/monotonous_counter.go

Reply:
```
counter := NewMonotonousCounter(0)

...
stored := counter.Set(height)
if stored {
	tc.maxReceiptHeight.Set(float64(height))
}

```